tests/unit/ui/components/TestShowMoreCard.js

```diff
@@ -1,105 +1,214 @@
+    const root = render({
+      children: 'Some text',
+    });
 
-    const contentNode = component.contents;
-    const truncateToMaxHeight = sinon.spy(component, 'truncateToMaxHeight');
-    root.setProps(); // simulate any kind of update to properties
+    const resetUIStateSpy = sinon.spy(root.instance(), 'resetUIState');
 
-    sinon.assert.calledWith(truncateToMaxHeight, contentNode);
+    root.setProps({
+      children: 'Some new text',
+    });
+
+    sinon.assert.called(resetUIStateSpy);
   });
 
-  it('expands if new child content is smaller', () => {
-    const root = mount(
-      <ShowMoreCardBase i18n={fakeI18n()}>
-        This would be very long content, but we cannot get `clientHeight` in the
-        tests, so this will be forced below (via setState()).
-      </ShowMoreCardBase>,
+  it("does not call resetUIState if the children's html is the same", () => {
+    /* eslint-disable react/no-danger */
+    const root = render({
+      children: (
+        <div
+          dangerouslySetInnerHTML={{
+            __html: '<span>Some component text.</span>',
+          }}
+        />
+      ),
+    });
+
+    const resetUIStateSpy = sinon.spy(root.instance(), 'resetUIState');
+
+    root.setProps({
+      children: (
+        <div
+          dangerouslySetInnerHTML={{
+            __html: '<span>Some component text.</span>',
+          }}
+        />
+      ),
+    });
+    /* eslint-enable react/no-danger */
+
+    sinon.assert.notCalled(resetUIStateSpy);
+  });
+
+  it("does not call resetUIState if the children's text is the same", () => {
+    const root = render({
+      children: 'Some text',
+    });
+
+    const resetUIStateSpy = sinon.spy(root.instance(), 'resetUIState');
+
+    root.setProps({
+      children: 'Some text',
+    });
+
+    sinon.assert.notCalled(resetUIStateSpy);
+  });
+
+  it('executes truncateToMaxHeight when it receives props changes', () => {
+    const root = render();
+
+    const contentNode = root.instance().contents;
+
+    const truncateToMaxHeight = sinon.spy(
+      root.instance(),
+      'truncateToMaxHeight',
     );
 
-    // We have to manually set the expanded flag to false because we don't have
-    // a `clientHeight` in the tests.
-    root.setState({ expanded: false });
+    // We are simulating any kind of update to properties.
+    root.setProps({ children: 'Some text' });
+
+    sinon.assert.calledWith(truncateToMaxHeight, contentNode);
+  });
+
+  it('does not execute truncateToMaxHeight when "read more" has been expanded', () => {
+    const { store } = dispatchClientMetadata();
 
-    expect(root.state('expanded')).toEqual(false);
-    // This will call `componentWillReceiveProps()`, the content of `children`
-    // is for example purpose.
-    root.setProps({ children: 'short content' });
-    expect(root.state('expanded')).toEqual(true);
+    // Simulating read more has been expanded already.
+    const root = render({
+      store: {
```

Comment:
I realize this contradicts what I said earlier, but I think the problem with what you're doing here - trying to render it with the correct state - will be fragile. The test shouldn't have to understand the internals of `uiState`. So maybe you do need to do some setup with the component in order to get it into the state you want for the start of the test.

Reply:
okay, I've set it up to do all the steps to get it to this point then